datalad/cmd.py

```diff
@@ -181,7 +181,7 @@ def run(self, cmd, log_stdout=True, log_stderr=True,
                                     shell=shell,
                                     cwd=cwd)
             except Exception, e:
-                lgr.error("Failed to start %s: %s" % (cmd, e))
+                lgr.error("Failed to start %r: %s" % (cmd, " under %r" % cwd if cwd else '', e))
```

Comment:
Ok -- I think I now see better what you originally meant -- as it is now, returned status is useless since it can't return anything but 0 since if not -- it just raises an exception.

Indeed we could simplify API and keep it more Pythonic I guess by coming up with an exception to throw which would include error code and output, while .run() would just always return output (thus we could deprecate return_output option).  If it returns without exception -- we already know it succeeded.